vkrunner/vr-script.c

```diff
@@ -1923,108 +1958,181 @@ process_token_replacements(struct load_state *data)
         return true;
 }
 
-static struct vr_script *
-load_script_from_stream(const struct vr_config *config,
-                        const char *filename,
-                        FILE *f)
+static struct load_state *
+load_script_begin(const struct vr_config *config,
+                  const char *filename)
 {
-        struct load_state data = {
-                .config = config,
-                .filename = filename,
-                .line_num = 1,
-                .script = vr_calloc(sizeof (struct vr_script)),
-                .line = VR_BUFFER_STATIC_INIT,
-                .buffer = VR_BUFFER_STATIC_INIT,
-                .commands = VR_BUFFER_STATIC_INIT,
-                .pipeline_keys = VR_BUFFER_STATIC_INIT,
-                .extensions = VR_BUFFER_STATIC_INIT,
-                .buffers = VR_BUFFER_STATIC_INIT,
-                .current_stage = -1,
-                .current_section = SECTION_NONE,
-                .clear_depth = 1.0f
-        };
-        bool res = true;
         int stage;
+        struct load_state *data = vr_calloc(sizeof (struct load_state));
```

Comment:
It might be nicer if load_state was allocated on the stack in the two load functions. Then load_script_begin could take a pointer to it.

Reply:
Done.